checkpoint 3 - attempt 1

zacharyehren opened the pull request.

zacharyehren committed:
checkpoint 3 - attempt 1

zacharyehren commented on app/pages/index.html:
```diff
@@ -29,5 +29,6 @@
   <script src="/scripts/app.js"></script>
   <script src="/scripts/services/Room.js"></script>
   <script src="/scripts/controllers/RoomsCtrl.js"></script>
+  <script src="/node_modules/angular-ui-bootstrap/src/modal/modal.js"></script>
```
I used a local path for this. Should it actually be a remote path? Also, should it be higher in the list of script tags? I know the order of the script tags depends on which tag is dependent on the others.

bmneely replied:
This should work fine. `npm` will install the modules into the `node_modules` directory just like you have them included.

zacharyehren committed:
Add flexbox to home.html to render column

bmneely commented on app/scripts/services/Room.js:
```diff
@@ -2,6 +2,13 @@
     var rooms = $firebaseArray(roomsRef);
+   
+    //My thoughts here were to first create a newRoom variable and set it to null since nothing has been created yet. Then within the function use the $add method to add the new room to the rooms array. I'm currently stuck on how to connect newRoom variable to the newRoom method in ModalCtrl.js
+    var newRoom = null;
```
You actually won't want to set this to `null` `newRoom` will only exist as function argument, not as a local variable in `Room`

bmneely commented on app/scripts/services/Room.js:
```diff
@@ -2,6 +2,13 @@
+    var newRoom = null;
+    var createNewRoom = function(){
+      rooms.$add(newRoom);
+    }
```
This is exactly, right. To make this a public function for your service, you'll want to add this on line 15 as part of your return:

```javascript
      all: rooms,
      create: function(newRoom) {
        rooms.$add(newRoom); 
      }
```

bmneely commented on dist/scripts/app.js:
```diff
@@ -12,10 +12,17 @@
       templateUrl: '/templates/home.html'
+    })
+    
+    //Do I need to inject the modal as a state? 
```
You do not, because the modal isn't a new route in your app (it'll be open within an existing page, not when a new page is navigated to.

zacharyehren committed:
fix newRoom function in Room.js, update ModalCtrl.js

bmneely committed:
Fix ui.bootstrap include

zacharyehren committed:
Merge pull request #3 from bmneely/checkpoint3-create-chat-rooms

Fix ui.bootstrap include

zacharyehren committed:
update ng-controller in home.html

bmneely committed:
Dependency injection fix

zacharyehren committed:
Merge pull request #4 from bmneely/checkpoint3-create-chat-rooms

Dependency injection fix

zacharyehren committed:
update template Url in ModalCtrl

zacharyehren committed:
Update ModalCtrl

Merge branch 'checkpoint3-create-chat-rooms' of https://github.com/zacharyehren/bloc-chat into checkpoint3-create-chat-rooms

zacharyehren committed:
update newRoom method + other updates

zacharyehren committed:
update ModalCtrl ModalInstanceCtrl & styles

bmneely committed:
Fix modal controller reference

zacharyehren committed:
Merge pull request #5 from bmneely/checkpoint3-create-chat-rooms

Fix modal controller reference

zacharyehren committed:
change the console test to an alert

zacharyehren committed:
checkpoint 3 complete
